modules/slugs/includes/qtranslate-slug-settings-options.php

```diff
@@ -47,15 +47,27 @@ function qts_options_page_fields() {
         $options[] = qts_options_page_build_slug_fields( $taxonomy, "taxonomies", "taxonomy_" );
     }
 
-    return $options;
+    return array_filter($options);
 }
 
 function qts_options_page_build_slug_fields( $object, $target_section, $id_prefix ) {
+    if ( ! is_array($object->rewrite) ) {
+        if ( $object->rewrite ){
+            $slug = $object->name;
+        }else{
+            return array();
```

Comment:
Could we have objects that don't have the `rewrite` field here? Shouldn't they be eliminated earlier?

Reply:
yes e.g. for posts and pages `rewrite==false` meaning that permalink bases cannot be modified. `rewrite`property can be boolean or array. as long as `rewrite!=false` permalink base is ok to be customized